[quant][fx][improvement] Renamed default_affine_fixed_qparams_observer and default_symmetric_fixed_qparams_observer

dzdang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #76637

Summary:
The previous naming convention `default_affine_fixed_qparams_observer`
and `default_symmetric_fixed_qparams_observer` were uninformative, and users had to read
the definition in order to understand what these observers are. The new
naming convention reveals information about the range of the observers

The analogous changes were also made for
`default_symmetric_fixed_qparams_fake_quant` and
`default_affine_fixed_qparams_fake_quant`

Test Plan:
```
python test/test_quantization.py
```

Differential Revision: [D36054169](https://our.internmc.facebook.com/intern/diff/D36054169)

dzdang committed:
[quant][fx][improvement] Renamed default_affine_fixed_qparams_observer and default_symmetric_fixed_qparams_observer

Summary:
The previous naming convention was uninformative, and users had to read
the definition in order to understand what these observers are. The new
naming convention reveals information about the range of the observers

Test Plan:
```
python test/test_quantization.py
```

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36054169).

dzdang committed:
Update on "[quant][fx][improvement] Renamed default_affine_fixed_qparams_observer and default_symmetric_fixed_qparams_observer"


Summary:
The previous naming convention was uninformative, and users had to read
the definition in order to understand what these observers are. The new
naming convention reveals information about the range of the observers

Test Plan:
```
python test/test_quantization.py
```

Differential Revision: [D36054169](https://our.internmc.facebook.com/intern/diff/D36054169)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36054169).

vkuzo commented on torch/ao/quantization/fake_quantize.py:
```diff
@@ -352,8 +352,12 @@ def forward(self, X: torch.Tensor) -> torch.Tensor:
 
-default_symmetric_fixed_qparams_fake_quant = FixedQParamsFakeQuantize.with_args(observer=default_symmetric_fixed_qparams_observer)
-default_affine_fixed_qparams_fake_quant = FixedQParamsFakeQuantize.with_args(observer=default_affine_fixed_qparams_observer)
+default_symmetric_fixed_qparams_fake_quant = (
```
would be good to change this name as well

dzdang replied:
will do

vkuzo commented on torch/ao/quantization/fake_quantize.py:
```diff
@@ -352,8 +352,12 @@ def forward(self, X: torch.Tensor) -> torch.Tensor:
+default_symmetric_fixed_qparams_fake_quant = (
+    FixedQParamsFakeQuantize.with_args(observer=default_symmetric_fixed_qparams_range_neg1to1_observer)
+)
+default_affine_fixed_qparams_fake_quant = (
```
would be good to change this name as well

vkuzo commented on torch/ao/quantization/observer.py:
```diff
@@ -1492,10 +1492,14 @@ def load_observer_state_dict(mod, obs_dict):
 # TODO(future PR): remove these defaults and enforce activation functions
 # to explicitly specify their output range
-default_symmetric_fixed_qparams_observer = FixedQParamsObserver.with_args(
+default_symmetric_fixed_qparams_range_neg1to1_observer = FixedQParamsObserver.with_args(
```
should we remove `symmetric` from the name, since `neg1to1` has a more clear version of the same information?

vkuzo commented on torch/ao/quantization/observer.py:
```diff
@@ -1492,10 +1492,14 @@ def load_observer_state_dict(mod, obs_dict):
+default_symmetric_fixed_qparams_range_neg1to1_observer = FixedQParamsObserver.with_args(
     scale=2.0 / 256.0, zero_point=128, dtype=torch.quint8, quant_min=0, quant_max=255)
-default_affine_fixed_qparams_observer = FixedQParamsObserver.with_args(
+default_affine_fixed_qparams_range_0to1_observer = FixedQParamsObserver.with_args(
```
should we remove `affine` from the name?

dzdang replied:
does range 0 to 1 imply it's affine quant? if not, and we remove affine, then I think the name could be ambiguous?

vkuzo replied:
usually in this codebase `affine` means "quantized values are not forced to be centered around zero", which is unrelated to how it is used in this variable name.  It would be cleaner just to delete `affine` from the variable name.

dzdang committed:
Update on "[quant][fx][improvement] Renamed default_affine_fixed_qparams_observer and default_symmetric_fixed_qparams_observer"


Summary:
The previous naming convention `default_affine_fixed_qparams_observer`
and `default_symmetric_fixed_qparams_observer` were uninformative, and users had to read
the definition in order to understand what these observers are. The new
naming convention reveals information about the range of the observers

The analogous changes were also made for
`default_symmetric_fixed_qparams_fake_quant` and
`default_affine_fixed_qparams_fake_quant`

Test Plan:
```
python test/test_quantization.py
```

Differential Revision: [D36054169](https://our.internmc.facebook.com/intern/diff/D36054169)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36054169).

dzdang committed:
Update on "[quant][fx][improvement] Renamed default_affine_fixed_qparams_observer and default_symmetric_fixed_qparams_observer"


Summary:
The previous naming convention `default_affine_fixed_qparams_observer`
and `default_symmetric_fixed_qparams_observer` were uninformative, and users had to read
the definition in order to understand what these observers are. The new
naming convention reveals information about the range of the observers

The analogous changes were also made for
`default_symmetric_fixed_qparams_fake_quant` and
`default_affine_fixed_qparams_fake_quant`

Test Plan:
```
python test/test_quantization.py
```

Differential Revision: [D36054169](https://our.internmc.facebook.com/intern/diff/D36054169)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36054169).

vkuzo reviewed (approved):
looks good! Some tests are failing, we should fix before landing.

dzdang committed:
Update on "[quant][fx][improvement] Renamed default_affine_fixed_qparams_observer and default_symmetric_fixed_qparams_observer"


Summary:
The previous naming convention `default_affine_fixed_qparams_observer`
and `default_symmetric_fixed_qparams_observer` were uninformative, and users had to read
the definition in order to understand what these observers are. The new
naming convention reveals information about the range of the observers

The analogous changes were also made for
`default_symmetric_fixed_qparams_fake_quant` and
`default_affine_fixed_qparams_fake_quant`

Test Plan:
```
python test/test_quantization.py
```

Differential Revision: [D36054169](https://our.internmc.facebook.com/intern/diff/D36054169)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36054169).
